### ISSUE TITLE
Missing dependencies: OpenCTM, Qhull

### ISSUE THREAD

**Issue by pnabbefeld** [bartoszek/AUR-meshlab-git#2]
Cannot install meshlab-git:

-- Searching for required components
-- Found OpenGL: /usr/lib/libOpenGL.so   
-- Searching for required components with bundled fallback
-- Found GLEW: /usr/include (found version "2.1.0") 
CMake Warning at CMakeLists.txt:75 (find_package):
  By not providing "FindEigen3.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Eigen3", but
  CMake did not find one.

  Could not find a package configuration file provided by "Eigen3" with any
  of the following names:

    Eigen3Config.cmake
    eigen3-config.cmake

  Add the installation prefix of "Eigen3" to CMAKE_PREFIX_PATH or set
  "Eigen3_DIR" to a directory containing one of the above files.  If "Eigen3"
  provides a separate development package or SDK, be sure it has been
  installed.

-- Searching for optional components
-- Found Lib3ds: /usr/include  
-- Found GMP: /usr/include (found version "6.2.0") 
-- Found muparser: /usr/include  
-- Could NOT find OpenCTM (missing: OPENCTM_INCLUDE_DIR OPENCTM_LIBRARY) 
-- Could NOT find Qhull (missing: QHULL_qhull_LIBRARY QHULL_qhull_INCLUDE_DIR qhull) 
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- - jhead - using bundled source
-- - glew - using system-provided library
CMake Error at external.cmake:68 (message):
  Eigen is required - at least one of ALLOW_SYSTEM_EIGEN or
  ALLOW_BUNDLED_EIGEN must be enabled and found.
Call Stack (most recent call first):
  CMakeLists.txt:119 (include)

**Comment by pnabbefeld** [comment]
OpenCTM does not seem to be available on Arch Linux, so it needs to be added - preferably as an additional package (so other builds could probably depend on it in the future).

I saw it's probably an optional dependency. However, it seems to be needed for 3DS support, which is why I wanted to install meshlab.

**Comment by bartoszek** [comment]
Both [OpenCTM](https://github.com/cnr-isti-vclab/meshlab/tree/master/src/external/OpenCTM-1.0.3) and [Qhull](https://github.com/cnr-isti-vclab/meshlab/tree/master/src/external/qhull-2003.1) are included in `meshlab` source tree.
I've already thought about scooping out `openctm` et al. https://github.com/bartoszek/AUR-meshlab/commit/9e3e6c1858f99760315189f0940742b710dbeef4

**Comment by pnabbefeld** [comment]
Hm, probably You could just remove the related dependency checks? As mentioned, it seems to be needed for 3DS support, which is what I need meshlab for.

**Comment by bartoszek** [comment]
>  You could just remove the related dependency checks?

There's no need, it's internally resolved by cmake script.
```
-- Could NOT find OpenCTM (missing: OPENCTM_INCLUDE_DIR OPENCTM_LIBRARY)
-- Could NOT find Qhull (missing: QHULL_qhull_LIBRARY QHULL_qhull_INCLUDE_DIR qhull)
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- - jhead - using bundled source
-- - glew - using bundled source
-- - Eigen - using bundled source
-- - newuoa - using bundled source
-- - levmar - using bundled source
-- - lib3ds - using system-provided library
-- - GMP/MPIR - using system-provided GMP library
-- - muparser - using system-provided library
-- - OpenCTM - using bundled source
-- - structure-synth - using bundled source
-- - qhull - using bundled source
...
[ 16% | 99<8<512 | 523.5bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/openctm.c.o
[ 16% | 100<8<511 | 527.6bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/stream.c.o
[ 16% | 101<8<510 | 521.8bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/compressRAW.c.o
[ 16% | 102<8<509 | 518.3bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/compressMG1.c.o
[ 16% | 103<8<508 | 516.8bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/compressMG2.c.o
[ 16% | 104<8<507 | 520.8bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/liblzma/Alloc.c.o
[ 16% | 105<8<506 | 519.0bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/liblzma/LzFind.c.o
[ 17% | 106<8<505 | 518.9bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/liblzma/LzmaDec.c.o
[ 17% | 107<7<505 | 523.3bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/liblzma/LzmaEnc.c.o
[ 17% | 108<6<505 | 521.8bps ] Building C object CMakeFiles/external-openctm.dir/external/OpenCTM-1.0.3/lib/liblzma/LzmaLib.c.o
[ 17% | 109<6<504 | 461.3bps ] Linking CXX static library libexternal-jhead.a
[ 17% | 110<5<504 | 442.6bps ] Linking CXX static library libexternal-qhull.a
[ 17% | 111<4<504 | 433.6bps ] Linking CXX static library libexternal-glew.a
[ 18% | 112<4<503 | 430.8bps ] Linking CXX static library libexternal-levmar.a
[ 18% | 113<3<503 | 416.7bps ] Linking CXX static library libexternal-openctm.a
[ 18% | 114<2<503 | 224.0bps ] Automatic MOC and UIC for target external-ssynth
```
Is thare a problem with `aur/lib3ds` not working with meshlab?

**Comment by bartoszek** [comment]
fixed: 05766e59208cb7f8dddbbde36c8d8d2be339ca9e
